perun-core/src/main/java/cz/metacentrum/perun/core/entry/UsersManagerEntry.java

```diff
@@ -454,6 +454,23 @@ public void removeUserExtSource(PerunSession sess, User user, UserExtSource user
 		getUsersManagerBl().removeUserExtSource(sess, user, userExtSource);
 	}
 
+	public void removeUserExtSource(PerunSession sess, User user, UserExtSource userExtSource, boolean forceDelete) throws InternalErrorException, UserNotExistsException, UserExtSourceNotExistsException, PrivilegeException, UserExtSourceAlreadyRemovedException {
```

Comment:
Implementation should use and check param ``forceDelete``. If true, it should continue, on false I suggest to simply call above method (without forced flag). Also I believe user shouldn't be able to force delete own UES (even if its available in API only), it should be available to Perun admins.